services/movie_session.py

```diff
@@ -0,0 +1,41 @@
+from datetime import datetime
+
+from django.db.models import QuerySet
+
+from db.models import Movie, CinemaHall, MovieSession
+
+
+def create_movie_session(movie_show_time: datetime,
+                         movie_id: int,
+                         cinema_hall_id: int
+                         ) -> None:
+    MovieSession.objects.create(show_time=movie_show_time,
+                                cinema_hall=cinema_hall_id,
+                                movie_id=movie_id
+                                )
+
+def get_movies_sessions(session_date: str = None
+                        ) -> QuerySet:
+    if session_date:
+        date = datetime.strptime(session_date, "%Y-%m-%d")
+        return MovieSession.objects.filter(show_time=date)
+    return MovieSession.objects.all()
+
+
+def get_movie_session_by_id(movie_session_id: int
+                            ) -> MovieSession:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(session_id: int,
+                         show_time: datetime = None,
+                         cinema_hall_id: int = None
+                         ) -> None:
+    sesh = MovieSession.objects.get(id=session_id)
+    if show_time:
+        sesh.show_time = show_time
+    if cinema_hall_id:
+        sesh.cinema_hall = cinema_hall_id
```

Comment:
Similar to line 13, you should assign a `CinemaHall` object to `sesh.cinema_hall` instead of `cinema_hall_id`. Fetch the `CinemaHall` object using the provided `cinema_hall_id`.